### PR TITLE
Add support for custom inference engine port

### PR DESCRIPTION
**Pull request by varungup90**
## Pull Request Description
[Please provide a clear and concise description of your changes here]

## Related Issues
Resolves: #1111 

**Important: Before submitting, please complete the description above and review the checklist below.**

---

<details>
<summary><strong>Contribution Guidelines (Expand for Details)</strong></summary>

<p>We appreciate your contribution to aibrix! To ensure a smooth review process and maintain high code quality, please adhere to the following guidelines:</p>

<h3>Pull Request Title Format</h3>
<p>Your PR title should start with one of these prefixes to indicate the nature of the change:</p>
<ul>
    <li><code>[Bug]</code>: Corrections to existing functionality</li>
    <li><code>[CI]</code>: Changes to build process or CI pipeline</li>
    <li><code>[Docs]</code>: Updates or additions to documentation</li>
    <li><code>[API]</code>: Modifications to aibrix's API or interface</li>
    <li><code>[CLI]</code>: Changes or additions to the Command Line Interface</li>
    <li><code>[Misc]</code>: For changes not covered above (use sparingly)</li>
</ul>
<p><em>Note: For changes spanning multiple categories, use multiple prefixes in order of importance.</em></p>

<h3>Submission Checklist</h3>
<ul>
    <li>[ ] PR title includes appropriate prefix(es)</li>
    <li>[ ] Changes are clearly explained in the PR description</li>
    <li>[ ] New and existing tests pass successfully</li>
    <li>[ ] Code adheres to project style and best practices</li>
    <li>[ ] Documentation updated to reflect changes (if applicable)</li>
    <li>[ ] Thorough testing completed, no regressions introduced</li>
</ul>

<p>By submitting this PR, you confirm that you've read these guidelines and your changes align with the project's contribution standards.</p>

</details>

### CODE REVIEW

**Comment by nwangfw** [comment]
Is the current approach support the multi-node case since there are multiple ports needed? For example, https://github.com/vllm-project/aibrix/blob/4acc1c062ca221d578b48f940e41e146ad7add44/samples/distributed/fleet-two-node.yaml#L59-L66

**Comment by varungup90** [comment]
> Does the current approach support the multi-node case? In multi-node inference, we may need to dynamically assign multiple ports. For example, https://github.com/vllm-project/aibrix/blob/4acc1c062ca221d578b48f940e41e146ad7add44/samples/distributed/fleet-two-node.yaml#L59-L66

For inference, only one port is required.

File: pkg/utils/pod.go [hunk 1]
```diff
@@ -238,3 +241,11 @@ func SelectRandomPod(pods []*v1.Pod, randomFn func(int) int) (*v1.Pod, error) {
 	randomPod := readyPods[randomFn(len(readyPods))]
 	return randomPod, nil
 }
+
+func GetModelPortForPod(pod *v1.Pod) int64 {
+	modelPort, err := strconv.ParseInt(pod.Labels[modelPortIdentifier], 10, 32)
```
Comment (googs1025) [inline pkg/utils/pod.go]: Should we add similar boundary conditions to make the code more robust?
🤔 
```go

   if pod == nil || pod.Labels == nil {
        return defaultPodMetricPort
    }

    portStr, exists := pod.Labels[modelPortIdentifier]
    if !exists {
        return defaultPodMetricPort
    }
```

Comment (varungup90) [reply pkg/utils/pod.go]: Current code already handles corner case scenarios, added more unit tests to cover them. Only missing is pod == nil check, which will not happen in routing use case.

Comment (Jeffwan) [reply pkg/utils/pod.go]: I think the current code can catch the error. even the label is not there, it return the value "". 

I recommend to add some warning logs here. If we do want to know whether it's missing label or invalid port value, we need to split the values.

labelValue, ok := pod.Labels[modelPortIdentifier]
if !ok {
    warn("label %q not found in pod", modelPortIdentifier)
}

Comment (varungup90) [reply pkg/utils/pod.go]: Problem with warning logs is that this method is invoked for each request, it will flood the logs. By default, 8000 will be picked up and if custom port is incorrectly set then for debugging purpose, targetpod ip & port are logged (with each request_id) and returned in response header as well. 

Other alternative is that we can add warning message for now as recommended by @googs1025 because its a new feature and help users to debug labels are correctly set. In future we can evaluate, whether to keep them or not.

Comment (varungup90) [reply pkg/utils/pod.go]: added a warn message if label is empty or incorrect value is set. Will re-evaluate later if need to keep the warn message.
